src/routeGeneration/templateHelpers.ts

```diff
@@ -362,6 +366,130 @@ export class ValidationService {
     return new Buffer(value);
   }
 
+  public validateUnion(
+    name: string,
+    value: any,
+    fieldErrors: FieldErrors,
+    minimalSwaggerConfig: SwaggerConfigRelatedToRoutes,
+    subSchemas: TsoaRoute.PropertySchema[] | undefined,
+    parent = '',
+  ): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors: FieldErrors[] = [];
+    const values: any[] = [];
+
+    subSchemas.forEach(subSchema => {
+      const subFieldError: FieldErrors = {};
+      const val = this.ValidateParam(subSchema, JSON.parse(JSON.stringify(value)), name, subFieldError, parent, minimalSwaggerConfig);
+      subFieldErrors.push(subFieldError);
+      values.push(val);
+    });
+
+    if (subFieldErrors.length > 0 && !subFieldErrors.some(subFieldError => Object.keys(subFieldError).length === 0)) {
+      fieldErrors[parent + name] = {
+        message: `Could not match the union against any of the items. Issues: ${JSON.stringify(subFieldErrors)}`,
+        value,
+      };
+      return;
+    }
+
+    if (value instanceof Object && minimalSwaggerConfig.noImplicitAdditionalProperties === 'silently-remove-extras') {
+      return values.reduce((acc, obj) => ({...acc, ...obj}), {});
+    }
+    return value;
+  }
+
+  public validateIntersection(
+    name: string,
+    value: any,
+    fieldErrors: FieldErrors,
+    swaggerConfig: SwaggerConfigRelatedToRoutes,
+    subSchemas: TsoaRoute.PropertySchema[] | undefined,
+    parent = '',
+  ): any {
+    if (!subSchemas) { return; }
```

Comment:
Shouldn't this assert/throw? I ask that because the metadata generation would have had to fail in order for an object to be created that has `dataType === 'union'`  or `dataType === 'union'`  but that doesn't have subSchemas, right?

Reply:
This will not ever happen, correct. However, TypeScript does not know that.

Reply:
Is there a way you would prefer to handle that? Throw a (specific) error?

Reply:
I was thinking something like:
```ts
throw new Error('internal tsoa error: the metadata that was generated should have had sub schemas since it’s for a intersection, however it did not. Please file an issue with tsoa at https://github.com/lukeautry/tsoa/issues’);
```
And then a similar one for in validateUnion.

The alternative is to use a [typeGuard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) outside of the function that narrows the type of the metadata to a type that definitely had subSchemas. But that’s a lot more work. I think throwing the error is fine since you pointed out that it will never be hit.

Reply:
https://github.com/lukeautry/tsoa/pull/400/commits/fb1f5231978893c73b3fcdd22118254f4bbf209c